platform/iphone/in_app_store.mm

```diff
@@ -116,167 +92,150 @@ - (void)productsRequest:(SKProductsRequest *)request didReceiveResponse:(SKProdu
-						receipt = [NSData dataWithContentsOfURL:receiptFileURL];
-						sdk_version = 7;
-
-					} else {
-						// Fall back to deprecated transaction receipt,
-						// which is still available in iOS 7.
-
-						// Use SKPaymentTransaction's transactionReceipt.
-						receipt = transaction.transactionReceipt;
-					}
-
-				} else {
-					receipt = transaction.transactionReceipt;
-				}
-
-				NSString *receipt_to_send = nil;
-				if (receipt != nil) {
-					receipt_to_send = [receipt description];
-				}
+				// Send the receipt back to Godot
 				Dictionary receipt_ret;
-				receipt_ret["receipt"] = String::utf8(receipt_to_send != nil ? [receipt_to_send UTF8String] : "");
-				receipt_ret["sdk"] = sdk_version;
+				receipt_ret["receipt"] = String::utf8(receipt != nil ? [receipt.description UTF8String] : "");
+				receipt_ret["sdk"] = 7;
 				ret["receipt"] = receipt_ret;
 
-				InAppStore::get_singleton()->_post_event(ret);
-
 				if (auto_finish_transactions) {
 					[[SKPaymentQueue defaultQueue] finishTransaction:transaction];
 				} else {
-					[pending_transactions setObject:transaction forKey:transaction.payment.productIdentifier];
+					pending_transactions[transaction.payment.productIdentifier] = transaction;
 				}
-
-			}; break;
+				break;
+			}
 			case SKPaymentTransactionStateFailed: {
 				printf("status transaction failed!\n");
-				String pid = String::utf8([transaction.payment.productIdentifier UTF8String]);
-				Dictionary ret;
-				ret["type"] = "purchase";
-				ret["result"] = "error";
-				ret["product_id"] = pid;
-				ret["error"] = String::utf8([transaction.error.localizedDescription UTF8String]);
-				InAppStore::get_singleton()->_post_event(ret);
 				[[SKPaymentQueue defaultQueue] finishTransaction:transaction];
-			} break;
+				break;
+			}
 			case SKPaymentTransactionStateRestored: {
 				printf("status transaction restored!\n");
-				String pid = String::utf8([transaction.originalTransaction.payment.productIdentifier UTF8String]);
 				InAppStore::get_singleton()->_record_purchase(pid);
-				Dictionary ret;
-				ret["type"] = "restore";
-				ret["result"] = "ok";
-				ret["product_id"] = pid;
-				InAppStore::get_singleton()->_post_event(ret);
 				[[SKPaymentQueue defaultQueue] finishTransaction:transaction];
-			} break;
+				break;
+			}
 			default: {
 				printf("status default %i!\n", (int)transaction.transactionState);
-			}; break;
-		};
-	};
-};
+				break;
+			}
+		}
+
+		// Send the variant back to Godot
+		InAppStore::get_singleton()->_post_event(ret);
+	}
+}
+
+#pragma mark - Helpers
+- (NSString *)localizedPrice:(SKProduct *)product {
+	NSNumberFormatter *numberFormatter = [self.class cachedPriceFormatter];
+	[numberFormatter setLocale:product.priceLocale];
+	return [numberFormatter stringFromNumber:product.price];
+}
+
++ (NSNumberFormatter *)cachedPriceFormatter {
+	static dispatch_once_t onceToken;
+	static NSNumberFormatter *gPriceFormatter = nil;
```

Comment:
The Apple Docs mention that it's very expensive to create `NSNumberFormatter` and `NSDateFormatter`. Here we're using dispatch to ensure this block is only run once for the life of the application. After that we'll simply update the `locale` prior to formatting each price to a string. 

This should save us on the performance cost, and has the added benefit of being lazily loaded.